ee-9/ee-9-api/src/main/resources/modules/system/layers/base/jakarta/activation/api/main/module.xml

```diff
@@ -25,7 +25,8 @@
 <module xmlns="urn:jboss:module:1.9" name="jakarta.activation.api">
 
     <resources>
-        <artifact name="${com.sun.activation:jakarta.activation}"/>
+        <artifact name="${jakarta.activation:jakarta.activation-api}"/>
+        <artifact name="${org.eclipse.angus:angus-activation}"/>
```

Comment:
Is putting API and impl in the same module the right thing to do?

I know in the past the answer was 'yes', but I'm wondering whether the fact they've now been split at Eclipse justifies a rethink.

Reply:
Probably not, because it leaves us with less flexibility.

@bstansberry I thought it was necessary to continue exporting to the callers the implementation which was already available under the `jakarta.activation.api` module. And the alternative to not add two artifacts to this module was to create a new module for the implementation, make it a dependency of `jakarta.activation.api` and export it, so the callers who depend on `jakarta.activation.api` don't get problems and can continue using the implementation classes from the caller classloader.

But that's conceptually wrong, the callers should use the API exposed by `jakarta.activation.api` and there is no need to export or reuse the implementation in the caller. The responsibility to handle the implementation classloader can be kept in the `jakarta.activation.api` module if we make the implementation a dependency of this module. There is no need to export the implementation. And I believe that callers should not be using classes from the implementation that are outside of the API contract.

So, yes, probably, this should be split into two modules.

Reply:
@yersan OK, lets try that. It would be good to do the split soon so if it causes problems we have time to notice and react. If we release an EE 10 server without a split we are somewhat constrained from doing the split later.